libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -134,691 +116,373 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
-        navigationEvent.totalStepCount = obtainStepCount(directionsRoute)
+        val directionsRoute = route ?: callbackDispatcher.routeProgress?.route
+        val location = newLocation ?: callbackDispatcher.lastLocation
+
+        navigationEvent.apply {
+            sdkIdentifier = this@MapboxNavigationTelemetry.sdkIdentifier
+
+            callbackDispatcher.routeProgress?.let { routeProgress ->
+                stepIndex = routeProgress.currentLegProgress?.currentStepProgress?.stepIndex ?: 0
+
+                distanceRemaining = routeProgress.distanceRemaining.toInt()
+                durationRemaining = routeProgress.durationRemaining.toInt()
+                distanceCompleted = routeProgress.distanceTraveled.toInt()
+
+                routeProgress.route.let {
+                    geometry = it.geometry()
+                    profile = it.routeOptions()?.profile()
+                    requestIdentifier = it.routeOptions()?.requestUuid()
+                    stepCount = obtainStepCount(it)
+                    legIndex = it.routeIndex()?.toInt() ?: 0
+                    legCount = it.legs()?.size ?: 0
+                }
+            }
+
+            callbackDispatcher.originalRoute.await().let {
+                originalStepCount = obtainStepCount(it)
+                originalEstimatedDistance = it.distance().toInt()
+                originalEstimatedDuration = it.duration().toInt()
+                originalRequestIdentifier = it.routeOptions()?.requestUuid()
+                originalGeometry = it.geometry()
+            }
+
+            locationEngine = locationEngineNameExternal
+            tripIdentifier = obtainUniversalUniqueIdentifier()
+            lat = location?.latitude ?: 0.0
+            lng = location?.longitude ?: 0.0
+            simulation = locationEngineNameExternal == MOCK_PROVIDER
+            percentTimeInPortrait = lifecycleMonitor?.obtainPortraitPercentage() ?: 100
+            percentTimeInForeground = lifecycleMonitor?.obtainForegroundPercentage() ?: 100
+
+            dynamicValues.let {
+                startTimestamp = generateCreateDateFormatted(it.sessionStartTime)
+                rerouteCount = it.rerouteCount.get()
+                sessionIdentifier = it.sessionId
+            }
+
+            eventVersion = EVENT_VERSION
+
+            directionsRoute?.let {
+                absoluteDistanceToDestination = obtainAbsoluteDistance(
+                    callbackDispatcher.lastLocation,
+                    obtainRouteDestination(it)
+                )
+                estimatedDistance = it.distance().toInt()
+                estimatedDuration = it.duration().toInt()
+
+                // TODO:OZ voiceIndex is not available in SDK 1.0 and was not set in the legacy telemetry        navigationEvent.voiceIndex
+                // TODO:OZ bannerIndex is not available in SDK 1.0 and was not set in the legacy telemetry        navigationEvent.bannerIndex
```

Comment:
Are these `TODO`s fixed?

Reply:
I don't know what to do with it. 
There are no `voiceIndex` or `bannerIndex` properties in the model classes.
Looks like can be removed.

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/a3dfcaa1622de207bb8a09e310f86a384c3af024/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/NavigationEvent.kt#L66-L67 Although it's true that in the legacy we were not setting them either https://github.com/mapbox/mapbox-navigation-android/blob/e232efd36700e4f5f3345bd76ad99cb28076bb1e/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationEvent.java#L369-L383 (these are optional). Maybe it's a good opportunity to start sending them. What do you think?

Reply:
yep, NavigationEvent has such fields, but what data should be used to init them?